### PR TITLE
fix: Don't fetch columns when fetching tables by default

### PR DESCRIPTION
**Pull request by ivasilov**
This PR changes the default behavior when fetching multiple tables. The columns are not included but they can be provided by adding a flag.

### CODE REVIEW

File: apps/studio/stores/pgmeta/MetaStore.ts [hunk 1]
```diff
@@ -426,7 +426,8 @@ export default class MetaStore implements IMetaStore {
     const connectionString = project?.connectionString
     const tables = await queryClient.fetchQuery({
       queryKey: tableKeys.list(projectRef, 'public'),
```
Comment (alaister) [inline apps/studio/stores/pgmeta/MetaStore.ts:428]: I'm not sure if this needs changing too?
```suggestion
      queryKey: tableKeys.list(projectRef, sourceTableSchema),
```

File: apps/studio/data/tables/tables-query.ts [hunk 2]
```diff
@@ -51,19 +64,19 @@ export async function getTables(
   if (Array.isArray(data) && sortByProperty) {
     return sortBy(data, (t) => t[sortByProperty]) as PostgresTable[]
   }
-  return data as PostgresTable[]
+  return data as Omit<PostgresTable, 'columns'>[]
```
Comment (alaister) [inline apps/studio/data/tables/tables-query.ts:67]: Columns will be included if `include_columns` is true though right? So wondering if using the `Omit` is correct?

File: apps/studio/data/tables/keys.ts [hunk 3]
```diff
@@ -1,6 +1,10 @@
 export const tableKeys = {
-  list: (projectRef: string | undefined, schema: string | undefined) =>
-    ['projects', projectRef, 'tables', { schema }] as const,
+  list: (
+    projectRef: string | undefined,
+    schema: string | undefined,
+    includeColumns?: boolean | undefined
+  ) =>
+    ['projects', projectRef, 'tables', { schema }, { includeColumns: !!includeColumns }] as const,
```
Comment (alaister) [inline apps/studio/data/tables/keys.ts:7]: Any reason you want another object in the key? I've always thought of the final object as kind of the "options" object.
```suggestion
    ['projects', projectRef, 'tables', { schema, includeColumns: !!includeColumns }] as const,
```

File: apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/ForeignKeySelector/ForeignKeySelector.tsx [hunk 4]
```diff
@@ -257,7 +259,7 @@ const ForeignKeySelector = ({
               ---
             </Listbox.Option>
             {/* @ts-ignore */}
```
Comment (alaister) [inline apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/ForeignKeySelector/ForeignKeySelector.tsx:261]: Wondering if this `@ts-ignore` can be removed while we're here?
```suggestion
```

**Review by alaister — COMMENTED** [review]
Looking good! Just a couple of questions:

File: apps/studio/data/tables/tables-query.ts (hunk 2 above)
Comment (ivasilov) [reply apps/studio/data/tables/tables-query.ts:67]: It returns no columns by default, so this type would be correct in most cases. When using `include_columns`, we would have to cast the return type ([example](https://github.com/supabase/supabase/pull/20157/files#diff-99d7ec12a697697fb966a710290a9603843572d25db8531694eccfce4090487fR67)). 

I don't have a strong opinion on this, just trying to make the default case more type-safe.

**Review by alaister — APPROVED** [review]
LGTM!

File: apps/studio/data/tables/tables-query.ts (hunk 2 above)
Comment (alaister) [reply apps/studio/data/tables/tables-query.ts:67]: I think casting is a viable solution for now – I don't think it's worth the time investment to get something "slightly" nicer 👍

**Pull request merged**